magprime/models.py

```diff
@@ -44,6 +44,31 @@ def child_to_attendee(self):
             self.badge_type = c.ATTENDEE_BADGE
             self.ribbon = remove_opt(self.ribbon_ints, c.UNDER_13)
 
+    def calculate_badge_cost(self, use_promo_code=True):
+        registered = self.registered_local if self.registered else None
+        base_badge_price = self.base_badge_price or c.get_attendee_price(registered)
+
+        if self.paid == c.NEED_NOT_PAY:
+            return 0
+        elif self.overridden_price is not None:
+            return self.overridden_price
+        elif self.is_dealer:
+            return c.DEALER_BADGE_PRICE
+        elif self.badge_type in [c.CHILD_BADGE, c.ATTENDEE_BADGE] and self.age_discount != 0:
+            return max(0, base_badge_price + self.age_discount)
+        elif self.badge_type in [c.ATTENDEE_BADGE, c.PSEUDO_GROUP_BADGE] and self.group \
```

Comment:
I don't _THINK_ this is actually needed but it doesn't hurt and I don't want to make this change live just to find out five minutes later that group discounts aren't working somehow :P